engine/src/test/java/com/datatorrent/stram/cli/ApexCliTest.java

```diff
@@ -180,4 +188,40 @@ public void testLaunchAppPackagePrecedenceWithConfigPackage() throws Exception
     Assert.assertEquals("app-default", props.get("dt.test.5"));
     Assert.assertEquals("package-default", props.get("dt.test.6"));
   }
+
+  @Test
+  public void testAppFromOnlyConfigPackage() throws Exception
+  {
+    ApexCli.LaunchCommandLineInfo commandLineInfo =
+        ApexCli.getLaunchCommandLineInfo(new String[]{"-conf", configFile.getAbsolutePath(), appFile.getAbsolutePath(), "-useConfigApps", "exclusive"});
+
+    ApexCli apexCli = new ApexCli();
+    apexCli.init();
+
+    Assert.assertEquals("configApps", "exclusive", commandLineInfo.useConfigApps);
+
+    apexCli.getLaunchAppPackageArgs(ap, cp, commandLineInfo, new ConsoleReader());
+
+    Assert.assertEquals(ap.getApplications().size(), 1);
+  }
+
+  @Test
+  public void testMergeAppFromConfigAndAppPackage() throws Exception
+  {
+    ApexCli.LaunchCommandLineInfo commandLineInfo =
+        ApexCli.getLaunchCommandLineInfo(new String[]{"-conf", configFile.getAbsolutePath(), appFile.getAbsolutePath(), "-useConfigApps", "inclusive"});
+
+    Assert.assertEquals("configApps", "inclusive", commandLineInfo.useConfigApps);
+
+    ApexCli apexCli = new ApexCli();
+    apexCli.init();
+
+    try {
+      apexCli.getLaunchAppPackageArgs(ap, cp, commandLineInfo, new ConsoleReader());
+    } catch (Throwable throwable ) {
```

Comment:
why are we catching throwable instead of the exception you expect it to throw? Also, there is an extra space before the closing paren.